src/main/webapp/app/overview/course-registration/course-registration.component.ts

```diff
@@ -0,0 +1,67 @@
+import { Component, OnInit } from '@angular/core';
+import { AccountService } from 'app/core/auth/account.service';
+import { Course } from 'app/entities/course.model';
+import { CourseManagementService } from 'app/course/manage/course-management.service';
+import { ProfileService } from 'app/shared/layouts/profiles/profile.service';
+import { matchesRegexFully } from 'app/utils/regex.util';
+import { HttpResponse } from '@angular/common/http';
+
+@Component({
+    selector: 'jhi-course-registration-selector',
+    templateUrl: './course-registration.component.html',
+})
+export class CourseRegistrationComponent implements OnInit {
+    courses: Course[];
+    public coursesToSelect: Course[] = [];
+    public userIsAllowedToRegister = false;
+    addedSuccessful = false;
+    loading = false;
+
+    constructor(private accountService: AccountService, private courseService: CourseManagementService, private profileService: ProfileService) {}
+
+    ngOnInit(): void {
+        this.loading = true;
+        this.courseService.findAllForDashboard().subscribe((res: HttpResponse<Course[]>) => {
+            this.courses = res.body!;
+            this.loadAndFilterCourses();
+        });
+        this.accountService.identity().then((user) => {
+            this.profileService.getProfileInfo().subscribe((profileInfo) => {
+                if (profileInfo) {
+                    this.userIsAllowedToRegister = matchesRegexFully(user!.login, profileInfo.allowedCourseRegistrationUsernamePattern);
+                }
+            });
+        });
+    }
+
+    trackCourseById(index: number, item: Course) {
+        return item.id;
+    }
+
+    loadAndFilterCourses() {
+        this.courseService.findAllToRegister().subscribe((registerRes) => {
```

Comment:
It is not the scope of the PR but since you changed this component quite a bit I think it's work mentioning. Maybe you can do it anyway or do it in a follow-up?

I think it would make more sense to directly return the courses filtered and sorted. We can optimise this call with a proper SQL command. As we have a specific endpoint for it anyway, that shouldn't be a problem. Once registering we can stick with removing the one course from the list.

It seems a bit odd to me to do two calls just to calculate a negative in the client if we can do it in one call without needing an additional special endpoint.

What do you think?

Reply:
@julian-christl Thanks for the review and the good points! I absolutely agree with all the remarks you made.
I focused on reducing the bundle sized and needed to refactor the component for that, but I just used what was there already.
I would do it in a follow up PR where I refactor the client and server-side and the tests, so that we can have the reduced bundle size as soon as possible

Reply:
Alright, thanks. Then please fix the Github actions issues. Seems like you got some unnecessary whitespace in there.

Also please assign me to review the follow-up.

Let's keep the conversations here open so we can late make sure that we know what was not yet handled.